tools/krane/build.rs

```diff
@@ -1,19 +1,51 @@
 use flate2::{read::GzDecoder, write::GzEncoder};
-use std::env;
 use std::fs::File;
 use std::io::{self, prelude::*};
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 use std::process::Command;
+use std::{env, fs};
 use tar::Archive;
 
 const CRANE_VERSION: &str = "0.20.1";
 
+fn apply_source_patches(source_path: impl AsRef<Path>, patch_dir: impl AsRef<Path>) {
+    let source_path = source_path.as_ref();
+    let patch_dir = patch_dir.as_ref();
+
+    which::which("patch").expect("Must have the `patch` utility installed in PATH");
+
+    let mut patches = fs::read_dir(patch_dir)
+        .expect("Failed to read patch directory")
+        .filter_map(|entry| entry.ok())
+        .map(|entry| entry.path())
+        .filter(|path| path.extension().map(|ext| ext == "patch").unwrap_or(false))
+        .collect::<Vec<_>>();
+    patches.sort();
+
+    for patch in patches {
+        println!("Executing `patch -p1 -i '{}'`", patch.display());
+
+        let patch_status = Command::new("patch")
+            .current_dir(source_path)
+            .arg("-p1")
+            .arg("-i")
+            .arg(patch.as_os_str())
+            .status()
+            .expect("Failed to execute patch command");
+
+        if !patch_status.success() {
+            panic!("Failed to apply patch '{}'", patch.display());
+        }
+    }
+}
+
 fn main() {
     let script_dir = env::current_dir().unwrap();
     let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
 
     println!("cargo::rerun-if-changed=../build-cache-fetch");
     println!("cargo::rerun-if-changed=hashes/crane");
+    println!("cargo::rerun-if-changed=patches");
```

Comment:
Interesting, I didn't realize `cargo` could watch an entire directory like this!